src/index.js

```diff
@@ -38,6 +38,7 @@ export default function pullWhen(sampler) {
 
       if (type === 1) {
         sink(1, data)
+        samplerTalkback(1, data)
```

Comment:
Before that line you should check if the source hasnt been unsubscribed. It's technically possible for a sink to unsubscribe after receiving that data and before `samplerTalkback` gets pulled.

Reply:
Good catch, thanks! I'll fix this soon.